server/jetstream_cluster.go

```diff
@@ -785,6 +808,21 @@ func (js *jetStream) monitorCluster() {
 			if n.Leader() {
 				js.checkClusterSize()
 			}
+		case <-lt.C:
+			s.Debugf("Checking JetStream cluster state")
+			// If we have a current leader or had one in the past we can cancel this here since the metaleader
+			// will be in charge of all peer state changes.
+			// For cold boot only.
+			if n.GroupLeader() != _EMPTY_ || n.HadPreviousLeader() {
+				lt.Stop()
+				continue
+			}
+			// If we are here we do not have a leader and we did not have a previous one, so cold start.
+			// Check to see if we can adjust our cluster size down iff we are in mixed mode and we have
+			// seen a total that is what our original estimate was.
+			if js, total := s.trackedJetStreamServers(); js < total && total >= n.ClusterSize() {
```

Comment:
I think given that this usually is a weird situation, and boot only, we need to be logging here at a level visible by default

Reply:
Agree, which level though? Just info?

Reply:
warn if we have such a level dont recall now

Reply:
I might suggest INFO level, since its probably a mixed mode setup by design. WDYT?

Reply:
Doesn’t it essentially only happen if the system is badly balanced? Potentially badly deployed? Not a normal situation?

but whatever, any log wil do given how little we have so whatever you wish.

Reply:
Originally I suggested that but if you recall we had a few folks chime in that that may be a normal setup where storage is only in a couple of clusters vs everywhere, so that could be a valid setup.

Reply:
ok pushed, pls check real quick.